### ISSUE TITLE
Baking `step_log()` without the requested column

### ISSUE THREAD

**Issue by DavisVaughan** [tidymodels/recipes#293]
Is it intentional that we don't error here? At the very least I think we should try and be consistent for what the steps do when the requested column isn't there.

``` r
suppressPackageStartupMessages(library(recipes))

iris_train <- iris[1:100,]
iris_test <- iris[101:150,]

rec <- recipe(Sepal.Width ~ Sepal.Length, iris_train) %>%
  step_log(Sepal.Width) %>%
  prep(iris_train)

iris_test_no_outcome <- subset(iris_test, select = -Sepal.Width)

bake(rec, iris_test_no_outcome)
#> # A tibble: 50 x 2
#>    Sepal.Length Sepal.Width
#>           <dbl>       <dbl>
#>  1          6.3          NA
#>  2          5.8          NA
#>  3          7.1          NA
#>  4          6.3          NA
#>  5          6.5          NA
#>  6          7.6          NA
#>  7          4.9          NA
#>  8          7.3          NA
#>  9          6.7          NA
#> 10          7.2          NA
#> # … with 40 more rows
```

<sup>Created on 2019-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

**Comment by DavisVaughan** [comment]
`tibble` throws an error for us in the `step_center()` case:

``` r
suppressPackageStartupMessages(library(recipes))

iris_train <- iris[1:100,]
iris_test <- iris[101:150,]

rec <- recipe(Sepal.Width ~ Sepal.Length, iris_train) %>%
  step_center(Sepal.Length) %>%
  prep(iris_train)

iris_test_no_outcome <- subset(iris_test, select = -Sepal.Length)

bake(rec, iris_test_no_outcome)
#> Error: Can't find column `Sepal.Length` in `.data`.
```

<sup>Created on 2019-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

**Comment by topepo** [comment]
This is what the [`skip` option](https://tidymodels.github.io/recipes/articles/Skipping.html) is for.

**Comment by juliasilge** [comment]
Related to #531 and #603

**Comment by juliasilge** [comment]
Closed in #813 🎉
